### DISCUSSION TITLE
prevent AlertDialogAction close the modal.

### DISCUSSION THREAD

**Discussion by m1guer** [shadcn-ui/ui#2911] [General]
is there a way to prevent the AlertDialogAction close the modal if the forms field are not filled?

**Comment by amaitari** [comment]
Change the AlertDialogAction type to "button" from "submit" since it's a button, and pass an onClick function to it and perform your validation in the handle submit function, this will prevent the alert dialog from closing, hope this helps

**Reply by Helios69** [reply]
Doesn't work.

**Comment by Salman9000** [comment]
did u find any solution
